### PR TITLE
Correct various behavior with organization settings

### PR DESCRIPTION
**Pull request by dcramer**
- Don't send lists with empty values
- Reject lists with empty values

#nochanges

### CODE REVIEW

**Comment by dcramer** [comment]
doing a quick QA pass on this yet

**Comment by dcramer** [comment]
confirmed this prevents empty values

File: src/sentry/static/sentry/app/views/organizationSettings.jsx [hunk 1]
```diff
@@ -82,8 +82,16 @@ const OrganizationSettingsForm = React.createClass({
...
         data: {
           ...formData,
-          safeFields: formData.safeFields.split('\n'),
-          sensitiveFields: formData.sensitiveFields.split('\n'),
+          safeFields: (
+            (formData.safeFields !== '') ?
+            formData.safeFields.split('\n') :
+            []
+          ),
+          sensitiveFields: (
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/organizationSettings.jsx]: This seems that it won't catch say, newlines at the end of input, etc. I see a few of these in the database. Or multiple just empty lines resulting in `[u'', u'']` in storage.

Comment (mattrobenolt) [inline src/sentry/static/sentry/app/views/organizationSettings.jsx]: Oh, it seems that the actual API response will block it though.

**Review by mattrobenolt — APPROVED** [review]

File: src/sentry/static/sentry/app/views/organizationSettings.jsx (hunk 1 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationSettings.jsx]: I'll add trim

File: src/sentry/static/sentry/app/views/organizationSettings.jsx [hunk 2]
```diff
@@ -82,8 +83,12 @@ const OrganizationSettingsForm = React.createClass({
         method: 'PUT',
         data: {
           ...formData,
-          safeFields: formData.safeFields.split('\n'),
-          sensitiveFields: formData.sensitiveFields.split('\n'),
+          safeFields: (
+            trim(formData.safeFields).split('\n').filter(f => f !== '')
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/organizationSettings.jsx]: Do we want to additionally trim *after* `split` (i.e. each row)?

**Comment by dcramer** [comment]
fyi someone can take this over and finish/land it as I have a bunch of commitments today

File: src/sentry/static/sentry/app/views/organizationSettings.jsx (hunk 2 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/organizationSettings.jsx]: I think more needs to be done here (this is using the regex from `utils.split`):

```javascript
'   derp      derp derp\n    derp derp\n      \ntest'
  .replace(/^\s+|\s+$/g,'').split('\n')
  .filter(f => f !== '');

// => ["derp      derp derp", "    derp derp", "      ", "test"]
```

**Comment by benvinegar** [comment]
> fyi someone can take this over and finish/land it as I have a bunch of commitments today

I got it.

**Pull request merged**